tests/test_modeling_tf_common.py

```diff
@@ -574,7 +591,10 @@ def check_hidden_states_output(config, inputs_dict, model_class):
                 self.model_tester, "expected_num_hidden_layers", self.model_tester.num_hidden_layers + 1
             )
 
-            hidden_states = outputs[-1]
+            if hasattr(outputs, "hidden_states"):
```

Comment:
can we have an `if model.config.is_encoder_decoder` case here instead ? Seems more in line with PT tests and we should test both `decoder_hidden_states` and `encoder_hidden_states` for encoder-decoder

Reply:
Better?